plugins/src/scripts/mod.rs

```diff
@@ -53,12 +54,27 @@ impl App {
     }
 
     async fn activate(&mut self, id: u32) {
+        let whitespace_re = Regex::new(r"[ \t]+").unwrap();
+
         if let Some(script) = self.scripts.get(id as usize) {
             let interpreter = script.interpreter.as_deref().unwrap_or("sh");
+
+            // split the shebang into parts, e.g. ["/bin/bash"], or a more complex ["/usr/bin/env", "bash"]
+            let parts: Vec<&str> = whitespace_re.split(interpreter).collect();
+
+            // first token must be the command to run, e.g. "/usr/bin/env"
+            let program: &str = parts.first().unwrap();
+
+            // keep args as given, e.g. "bash"
+            let mut args: Vec<&str> = parts[1..].to_vec();
+
+            // finally, add the script file itself as a final arg
+            args.push(script.path.to_str().unwrap());
+
             send(&mut self.out, PluginResponse::Close).await;
 
-            let _ = Command::new(interpreter)
-                .arg(script.path.as_os_str())
+            let _ = Command::new(program)
```

Comment:
Huh, is there a reason why we don't use `Command::new(script.path)` instead and rely on the OS/shell/binfmt/whatever to execute the script? That way we don't necessarily even need to parse the sha-bang line at all

Reply:
I'm fairly new to Rust, but I read the `Command::new` docs as saying the string passed in must be either an abs path to the executable, or the command to search for on the path, with no args:

```
Constructs a new Command for launching the program at path program, with the following default configuration:

No arguments to the program
...
```

https://doc.rust-lang.org/std/process/struct.Command.html

Reply:
Oh, I see what you're saying. Just pass the script itself in. Maybe that would work!

Reply:
Maybe, but there's probably a reason why the original code worked the way it did, and it old code does have the benefit of only needing read-permissions on the script file :shrug: